server/aws-lsp-codewhisperer/src/language-server/agenticChat/agenticChatController.ts

```diff
@@ -229,18 +232,20 @@ export class AgenticChatController implements ChatHandlers {
         chatSessionManagementService: ChatSessionManagementService,
         features: Features,
         telemetryService: TelemetryService,
-        serviceManager?: AmazonQTokenServiceManager
+        serviceManager?: AmazonQBaseServiceManager
     ) {
         this.#features = features
         this.#chatSessionManagementService = chatSessionManagementService
         this.#triggerContext = new AgenticChatTriggerContext(features)
         this.#telemetryController = new ChatTelemetryController(features, telemetryService)
         this.#telemetryService = telemetryService
         this.#serviceManager = serviceManager
-        this.#serviceManager?.onRegionChange(region => {
-            // @ts-ignore
-            this.#features.chat.chatOptionsUpdate({ region })
-        })
+        if (this.#serviceManager instanceof AmazonQTokenServiceManager) {
```

Comment:
please implement the onRegionChange in the serviceManager. In the AmazonQBaseServiceManager and AmazonQIAMServiceManager, do nothing 

Avoid having this `if` everywhere.